clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1053,6 +1054,7 @@ void EmitAssemblyHelper::EmitAssemblyWithLegacyPassManager(
     if (CodeGenOpts.DisableLLVMPasses)
       PerModulePasses.add(createAlwaysInlinerLegacyPass(false));
     PerModulePasses.add(createSYCLLowerWGLocalMemoryLegacyPass());
+    PerModulePasses.add(createSYCLMutatePrintfAddrspaceLegacyPass());
```

Comment:
Could you please also add this pass into new pass manager's pipeline (see commit https://github.com/intel/llvm/pull/4860/commits/07e718be082f0546cdff9e81fd9cdb1775f93fc8)?

Reply:
I've fetched the latest revision of the `sycl` branch, but the changes you've pointed at don't seem to have arrived there yet. Should I wait for the merge to incorporate these, possibly as a follow-up PR?

Reply:
I think that you can just add the following code:
```
      if (LangOpts.SYCLIsDevice) {
        MPM.addPass(SYCLMutatePrintfAddrspacePass());
      }
```
into EmitAssemblyHelper::RunOptimizationPipeline function right before VerifierPass is added.

Reply:
Done just like that, thanks!